### PR TITLE
Improve nvdim documentation

### PR DESCRIPTION
**Pull request by fangohr**
based on #271, improve `nvdim` related documentation: make clear we mean the NUMBER of dimensions (i.e. an int, and not the names of the dimensions.)

### CODE REVIEW

File: discretisedfield/field.py [hunk 1]
```diff
@@ -41,8 +41,8 @@ class Field(_FieldIO):
 
     nvdim : int
 
-        Dimension of the field's value. For instance, if `dim=3` the field is a
-        three-dimensional vector field and for `dim=1` the field is a scalar
+        Number of Value DIMensions of the field. For instance, if `nvdim=3` the field is a
```
Comment (samjrholt) [inline discretisedfield/field.py:44]: Is DIM capitalised on purpose?
Same with Value

File: discretisedfield/field.py [hunk 2]
```diff
@@ -1401,11 +1401,11 @@ def __mul__(self, other):
 
         It can be applied between:
 
-        1. Two fields with equal vector dimentions,
+        1. Two fields with equal vector dimensions ``nvdim``,
```
Comment (samjrholt) [inline discretisedfield/field.py:1404]: We seem to use a mixture of `value dimensions` and `vector dimensions` when describing `nvdims`.

**Review by samjrholt — CHANGES_REQUESTED** [review]

File: discretisedfield/field.py (hunk 1 above)
Comment (fangohr) [reply discretisedfield/field.py:44]: Yes - the idea is that the capitalised letters form the argument name (`nvdim`) together.

File: discretisedfield/field.py (hunk 2 above)
Comment (fangohr) [reply discretisedfield/field.py:1404]: Yes, that is a bit unfortunate. We mean value dimensions but in this particular example those values are vector fields, so the 'v' can be misread. I had noticed but didn't see a good solution.

Perhaps we can get rid of the `vector` here (and potentially replace it by `value`): if both fields are scalar, we can also multiply them?

**Review by samjrholt — APPROVED** [review]

**Pull request merged**